### ISSUE TITLE
[BUG] Multiple targets in ispc shows not disassembly

### ISSUE THREAD

**Issue by tsafin** [compiler-explorer/compiler-explorer#2715]
**Describe the bug**
If we leave multiple ispc targets (as it usually used in practice) then we end up with empty disassembly.

**To Reproduce**

i.e. there is no disassembly for 
`--pic --target=avx2-i32x8,avx512skx-i32x16 --arch=x86-64`
but there is for 
`--pic --target=avx2-i32x8 --arch=x86-64` in command-line options

**Expected behavior**
Assume that despite of multiple targets used we should still see (at least) something default.

**Link**
https://godbolt.org/z/6nWPd36fn

Assumption is that it's triggered by multiple binaries generated by ispc in this case, and we had to deal with multiple, not single binary.

**Comment by dbabokin** [comment]
That's right, with multiple target specified ISPC produces multiple object files / assembly files. N+1 output files are generated. I.e. for `--target=avx2-i32x8,avx512skx-i32x16` there will be 3 output files:
1. original (the one that has name specified in `-o` switch`) output file, which will contain dispatch code.
2. file with `_avx2` suffix, i.e. for `-o out.ll` it will be `out_avx2.ll`
3. file with `_avx512` suffix, i.e. for `-o out.ll` it will be `out_avx512.ll`

I'm not sure why Compile Explorer is not showing the dispatch file, but anyway it's not what user would expect to see.

I guess the right fix is to change ISPC to produce a single object file / assembly with all the code for all platforms. It's in our plans, but I don't have exact ETA.

**Comment by mattgodbolt** [comment]
There's a somewhat abandoned PR that might be relevant (I'm not an expert, so apologies if I'm conflating) -> #2306 -- is there something in this that can be repurposed?

**Comment by dbabokin** [comment]
I've tried looking at this PR, but my JS knowledge is not sufficient to understand what's going on there...

**Comment by mattgodbolt** [comment]
@pbrubaker do you have any thoughts on this? Or any way of contacting @jcranmer-intel ? AIUI the heterogenous output showed multile targets' outputs.

**Comment by pbrubaker** [comment]
Oh my!  My apologies for not seeing this sooner.  Dmitry is right on about what's going on.  CE did show the dispatch file at one point though, as I used it to figure out how the dispatching worked.

I think the only thing you could probably do for this would be to show multiple object files?  Maybe a tabbed output window or something?  But that would be a lot of work, especially for only one compiler that has this behavior.

When I use CE with ISPC I just don't specify multiple targets, or I have a compiler window open for each target.

**Comment by mattgodbolt** [comment]
I've since merged in and understoof @jcranmer-intel 's code - it's now on the live site. So this is available to us: we parse out and separate the different ISA's outputs and show them with a dropdown. I'm sure we can get a similar output to work here (if not the exact same output). It's the "Device output" that appears when we detect multiple ISAs: https://godbolt.org/z/b67Y57oEs for example.

If there's a similar thing to leverage here, that'd be cool!

**Comment by pbrubaker** [comment]
Hey that's pretty awesome.  That would honestly be pretty cool to do with ISPC.  Does the device viewer pane just pop up automatically?

**Comment by mattgodbolt** [comment]
It does, if it discovers multiple outputs (which in the case above all appear in the same output file, and are detected by a magic regex). We would need to teach the system to find multiple outputs, and run and process those all too somehow. But the UI work is mostly there, it seems

**Comment by dbabokin** [comment]
Using trunk is fine. But `knl` target should be removed from the list, as it was deprecated and removed.

trunk version without knl: https://godbolt.org/z/1Tvb9Kqje
